mono/mini/mini-trampolines.c

```diff
@@ -732,10 +732,6 @@ common_call_trampoline (mgreg_t *regs, guint8 *code, MonoMethod *m, MonoVTable *
 		need_rgctx_tramp = FALSE;
 	}
 
-	/* Calls made through delegates on platforms without delegate trampolines */
-	if (!code && mono_method_needs_static_rgctx_invoke (m, FALSE))
-		need_rgctx_tramp = TRUE;
-
```

Comment:
There is a reason this code is there, why is it removed ?

Reply:
It breaks things, and does not seem to fix anything -- still to run a full CI/test pass.

The comment is about delegates, but it actually triggers for jmp trampolines, and appears to trigger for nothing else, including not for delegates.

From what I see, the only time code is null, is for jump trampolines.
If we get here for a call, or anything else, this code does nothing.
If we get here for a jump, this code sometimes sets need_rgctx_tramp to true and that breaks things. I don't fully understand what this trying to do vs. what is already done, but I have a vague sense that this is redundant, and then less clearly, does the wrong thing.

In any case, there are tests now..and either this is the right fix, or something is wrong elsewhere or nearby.

Reply:
Why does this break things ?

Reply:
It breaks the tests and F#, if you allow tailcall with vtable_arg.
I didn't determine what is the new parameter in r10 or r11. What was already coming in was correct and works, w/o this code changing things.
I noticed 1) it is where tailcall vs. non-tailcall diverge and 2) it changes if code works or not.